opm/autodiff/BlackoilModelBase.hpp

```diff
@@ -269,11 +269,42 @@ namespace Opm {
             ADB              mob;   // Phase mobility (per cell)
         };
 
-        struct WellOps {
-            WellOps(const Wells* wells);
-            Eigen::SparseMatrix<double> w2p;              // well -> perf (scatter)
-            Eigen::SparseMatrix<double> p2w;              // perf -> well (gather)
-            std::vector<int> well_cells;                  // the set of perforated cells
+        class StandardWells {
+        protected:
+            struct WellOps {
+                WellOps(const Wells* wells);
+                Eigen::SparseMatrix<double> w2p;              // well -> perf (scatter)
+                Eigen::SparseMatrix<double> p2w;              // perf -> well (gather)
+                std::vector<int> well_cells;                  // the set of perforated cells
+            };
+
+        public:
+            StandardWells(const Wells* wells);
+
+            const Wells& wells() const;
+
+            // return true if wells are available in the reservoir
+            bool wellsActive() const;
+            bool& wellsActive();
```

Comment:
This overload returning a bool reference looks strange, should probably be removed.